### PR TITLE
<rdar://problem/48765726> Allow generate-xcodeproj to generate static…

### PR DESCRIPTION
**Pull request by alblue**
… libraries

### CODE REVIEW

**Comment by alblue** [comment]
@swift-ci please smoke test

File: Sources/PackageModel/Target.swift [hunk 1]
```diff
@@ -11,12 +11,53 @@
...
+                case .static:
+                    return "static"
+                }
+            case .systemModule:
+                return "system-target"
+            }
+        }
+        public var isLibrary: Bool {
+            switch self {
+            case.library(_):
```
Comment (jakepetroules) [inline Sources/PackageModel/Target.swift]: `(_)` can be omitted

File: Sources/PackageModel/Target.swift [hunk 2]
```diff
@@ -11,12 +11,53 @@
...
     /// The target kind.
-    public enum Kind: String {
+    public enum Kind: CustomStringConvertible, Equatable {
         case executable
-        case library
-        case systemModule = "system-target"
+        case library(LibraryType)
+        case systemModule
         case test
+        public var description: String {
```
Comment (jakepetroules) [inline Sources/PackageModel/Target.swift]: `description` is not intended for stable results i.e. those used in serialization formats. Also, the documentation explicitly states:

    Calling this property directly is discouraged.

Please return to `rawValue`.

Comment (alblue) [reply Sources/PackageModel/Target.swift]: When I made the change, the swift compiler complained that you can't have a `rawValue` in conjunction with an enum with cases. In any case, this code was copied verbatim from `ProductType` in https://github.com/apple/swift-package-manager/blob/master/Sources/PackageModel/Product.swift#L17-L53

File: Sources/PackageModel/Target.swift (hunk 1 above)
Comment (alblue) [reply Sources/PackageModel/Target.swift]: Done

**Comment by alblue** [comment]
@swift-ci please smoke test

File: Sources/PackageModel/Target.swift (hunk 2 above)
Comment (jakepetroules) [reply Sources/PackageModel/Target.swift]: I didn't mean the automatically synthesized implementation - you can define `rawValue` yourself (or call it something else).

**Comment by alblue** [comment]
@swift-ci please test

**Comment by alblue** [comment]
@swift-ci please smoke test

**Comment by alblue** [comment]
Seems to have failed with the same reason I saw when I changed it earlier. Instead of using rawValue, do you want me to revert to the previous version, but just change 'description' to a different name instead?

**Comment by jakepetroules** [comment]
Just remove the inheritance from `String`. For example, this works:

    enum Foo {
        case bar
        case baz(Int)

        var rawValue: String {
            switch self {
            case .bar:
                return "Bar"
            case let .baz(a):
                return "Baz\(a)"
            }
        }
    }

**Comment by alblue** [comment]
@swift-ci please smoke test

**Comment by alblue** [comment]
@swift-ci please smoke test

**Comment by alblue** [comment]
Build failed with:

11:00:35 <unknown>:0: warning: missing submodule 'SwiftOverlayShims'
11:00:35 
/home/buildnode/jenkins/workspace/swift-package-manager-Linux-smoke-test/swiftpm/Sources/PackageDescription4/Package.swift:12:8: error: missing required module 'SwiftOverlayShims'
11:00:35 import Glibc
11:00:35        ^

**Comment by alblue** [comment]
@swift-ci please smoke test

**Comment by alblue** [comment]
@swift-ci please smoke test

File: Sources/PackageModel/Target.swift [hunk 3]
```diff
@@ -11,12 +11,53 @@
 import Basic
 
 public class Target: ObjectIdentifierProtocol {
+
+    /// The type of library.
+    public enum LibraryType: String, Codable {
```
Comment (aciidb0mb3r) [inline Sources/PackageModel/Target.swift]: Hmm, I am not sure this is correct way to model this. Targets are not static/dynamic/automatic, products are.

Comment (alblue) [reply Sources/PackageModel/Target.swift]: How do you recommend modelling this? The generate-xcodeproj uses the `Target` to generate its dependencies ...

https://github.com/apple/swift-package-manager/blob/master/Sources/Xcodeproj/Target+PBXProj.swift#L38-L48

**Pull request closed without merging**